sdk/platform/http_client/curl/src/az_curl.c

```diff
@@ -332,14 +328,15 @@ _az_http_client_curl_send_post_request(CURL* p_curl, _az_http_request const* p_r
   AZ_PRECONDITION_NOT_NULL(p_request);
 
   // Method
+  az_span request_body = { 0 };
+  AZ_RETURN_IF_FAILED(az_http_request_get_parts(p_request, NULL, NULL, &request_body));
```

Comment:
@JeffreyRichter - should these be "getters" instead of a single method call with 3-4 branches, especially because it isn't clear when this method would ever fail?
```C
AZ_NODISCARD az_http_method az_http_request_get_method(p_request);
AZ_NODISCARD az_span az_http_request_get_url(p_request);
AZ_NODISCARD az_span az_http_request_get_body(p_request);
```

Reply:
I would be fine with any of those. Probably easier to learn just one function like it is now than 3.
I believe this function is expecting transport adapters to get all http_request parts easily with just one line and have them copied as how the http stack uses them. For how libcurl is implemented, we would find it easier with getters since we never need the 3 values on a same method.
But other http stacks might find it easier in one call

Reply:
It's not a big deal either way. First, only Transport implementors need these values so it's not an API that need to be simple/easy to call. Victor is right in that I was trying to minimize the number of functions transport implementors need to know about and use. And, all transport implementors MUST use all three values so it made sense to me to have a single function, that returns all 3 values as they were at a single moment in time.

If the curl transport implementation has to call the 1 function that returns all 3 values in multiple places, then the benefit is probably lost and so 3 separate functions might be better then.

Reply:
> If the curl transport implementation has to call the 1 function that returns all 3 values in multiple places, then the benefit is probably lost and so 3 separate functions might be better then.

Agreed. We currently call it in 4 different places. This might just be an artifact of how we have implemented the curl transport (which we may be able to refactor), but based on what we have, independent getters make more sense.